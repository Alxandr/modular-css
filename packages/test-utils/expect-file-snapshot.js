"use strict";

const fs = require("fs");

const { toMatchSnapshot } = require("jest-snapshot");

expect.extend({
    toMatchFileSnapshot(source, ...args) {
        return toMatchSnapshot.call(
            this,
            fs.readFileSync(source, "utf8"),
            ...args,
        );
    },
});
