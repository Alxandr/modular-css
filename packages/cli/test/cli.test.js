"use strict";

const tester = require("cli-tester");

const { temp } = require("test-utils/fixtures.js");

require("test-utils/expect-file-snapshot.js");

const cli = require.resolve("../cli.js");

describe("/cli.js", () => {
    it("should show help with no args", async () => {
        const out = await tester(
            cli,
        );

        expect(out.code).toBe(2);
        expect(out.stdout).toMatchSnapshot();
    });

    it("should default to outputting to stdout", async () => {
        const out = await tester(
            cli,
            "fixtures/simple.css",
        );

        expect(out.code).toBe(0);
        expect(out.stdout).toMatchSnapshot();
    });

    it("should support outputting to a file (--out)", async () => {
        const out = await tester(
            cli,
            `--out=${temp("simple.css")}`,
            "fixtures/simple.css",
        );

        expect(out.code).toBe(0);
        expect(temp("simple.css")).toMatchFileSnapshot();
    });

    it("should support outputting compositions to a file (--json)", async () => {
        const out = await tester(
            cli,
            `--json=${temp("classes.json")}`,
            "fixtures/simple.css",
        );

        expect(out.code).toBe(0);
        expect(temp("classes.json")).toMatchFileSnapshot();
    });

    it("should return the correct error code on invalid CSS", async () => {
        const out = await tester(
            cli,
            "fixtures/invalid.css",
        );

        expect(out.code).toBe(1);
        expect(out.stderr).toMatch("Invalid composes reference");
    });

    it("should support disabling url() rewriting (--no-rewrite)", async () => {
        const out = await tester(
            cli,
            "--no-rewrite",
            `--out=${temp("no-rewrite.css")}`,
            "fixtures/no-rewrite.css",
        );

        expect(out.code).toBe(0);
        expect(temp("no-rewrite.css")).toMatchFileSnapshot();
    });
});
