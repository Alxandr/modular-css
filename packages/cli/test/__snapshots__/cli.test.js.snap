// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/cli.js should default to outputting to stdout 1`] = `
"/* fixtures/simple.css */
.mcc92a931a_fooga {
    color: red;
}"
`;

exports[`/cli.js should show help with no args 1`] = `
"CLI for modular-css

  Usage
  $ modular-css [options] <glob>...

  Options
  --dir,     -d <dir>    Directory to search from [process cwd]
  --out,     -o <file>   File to write output CSS to [stdout]
  --json,    -j <file>   File to write output compositions JSON to
  --map,     -m          Include inline source map in output
  --rewrite, -r          Control rewriting of url() references in CSS
  --help                 Show this help"
`;

exports[`/cli.js should support disabling url() rewriting (--no-rewrite) 1`] = `
"/* fixtures/no-rewrite.css */
.mc91ebdec9_a {
    background: url(\\"image.png\\");
}
"
`;

exports[`/cli.js should support outputting compositions to a file (--json) 1`] = `
"{
    \\"fixtures/simple.css\\": {
        \\"str\\": \\"\\\\\\"string\\\\\\"\\",
        \\"fooga\\": \\"mcc92a931a_fooga\\"
    }
}"
`;

exports[`/cli.js should support outputting to a file (--out) 1`] = `
"/* fixtures/simple.css */
.mcc92a931a_fooga {
    color: red;
}
"
`;
